packages/diagrams/frontend/sirius-components-diagrams/src/renderer/tools/useSingleClickTool.tsx

```diff
@@ -0,0 +1,173 @@
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { gql, useMutation } from '@apollo/client';
+import { GQLErrorPayload, useMultiToast, useSelection } from '@eclipse-sirius/sirius-components-core';
+import { useDialog } from '../../dialog/useDialog';
+import { useImpactAnalysisDialog } from '../palette/impact-analysis/useImpactAnalysisDialog';
+import { GQLSingleClickOnDiagramElementTool, GQLTool } from '../palette/Palette.types';
+import {
+  GQLInvokeSingleClickOnDiagramElementToolData,
+  GQLInvokeSingleClickOnDiagramElementToolInput,
+  GQLInvokeSingleClickOnDiagramElementToolPayload,
+  GQLInvokeSingleClickOnDiagramElementToolSuccessPayload,
+  GQLInvokeSingleClickOnDiagramElementToolVariables,
+  GQLToolVariable,
+  UseSingleClickToolValue,
+} from './useSingleClickTool.types';
+
+const invokeSingleClickOnDiagramElementToolMutation = gql`
+  mutation invokeSingleClickOnDiagramElementTool($input: InvokeSingleClickOnDiagramElementToolInput!) {
+    invokeSingleClickOnDiagramElementTool(input: $input) {
+      __typename
+      ... on InvokeSingleClickOnDiagramElementToolSuccessPayload {
+        newSelection {
+          entries {
+            id
+          }
+        }
+        messages {
+          body
+          level
+        }
+      }
+      ... on ErrorPayload {
+        messages {
+          body
+          level
+        }
+      }
+    }
+  }
+`;
+
+const isErrorPayload = (payload: GQLInvokeSingleClickOnDiagramElementToolPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+const isInvokeSingleClickSuccessPayload = (
+  payload: GQLInvokeSingleClickOnDiagramElementToolPayload
+): payload is GQLInvokeSingleClickOnDiagramElementToolSuccessPayload =>
+  payload.__typename === 'InvokeSingleClickOnDiagramElementToolSuccessPayload';
+
+const isSingleClickOnDiagramElementTool = (tool: GQLTool): tool is GQLSingleClickOnDiagramElementTool =>
+  tool.__typename === 'SingleClickOnDiagramElementTool';
+
+export const useSingleClickTool = (): UseSingleClickToolValue => {
+  const { addMessages } = useMultiToast();
+  const { setSelection } = useSelection();
+  const { showDialog } = useDialog();
+  const { showImpactAnalysisDialog } = useImpactAnalysisDialog();
+
+  const [invokeSingleClickOnDiagramElementTool] = useMutation<
+    GQLInvokeSingleClickOnDiagramElementToolData,
+    GQLInvokeSingleClickOnDiagramElementToolVariables
+  >(invokeSingleClickOnDiagramElementToolMutation);
+
+  const invokeSingleClickTool = async (
+    editingContextId: string,
+    diagramId: string,
+    tool: GQLTool,
```

Comment:
I think that all the code paths that lead to this, have a tool with the type `GQLSingleClickOnDiagramElementTool` so the check line 84 is not necessary anymore.